src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,95 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private int capacity;
+    private int threshold;
+    private int size;
+    private Node<K, V>[] array;
 
     @Override
     public void put(K key, V value) {
-
+        if (capacity == 0) {
+            array = (Node<K, V>[]) new Node[INITIAL_CAPACITY];
+            capacity = INITIAL_CAPACITY;
+            threshold = (int) (capacity * LOAD_FACTOR);
```

Comment:
Yes, it could make things nicer.
Been told not to create constructors where they aren't necessary. Not sure why, have to ask
Until then, I'll stick to my version)